superset-frontend/src/components/Select/Select.tsx

```diff
@@ -260,10 +294,110 @@ const Select = forwardRef(
       }
     }, [isLoading, loading]);
 
+    const selectAllMode = useMemo(
+      () => ensureIsArray(selectValue).length === fullSelectOptions.length + 1,
+      [selectValue, fullSelectOptions],
+    );
+
     useEffect(() => {
-      setSelectValue(value);
+      // if all values are selected, add select all to value
+      if (
```

Comment:
added isSingleMode and labelInValue as dependencies, but i purposefully left out fullSelectOptions.length because this hook should only be called when the component props have been changed. fullSelectOptions can change when new options are added and in that case we don't want the select value to be reset (erasing the newly added options).

Reply:
Good point. This is indicative that the effect should be broken into two unrelated operations. Like this:

```
useEffect(() => {
  setSelectValue(value);
}, [value]);

useEffect(() => {
  // if all values are selected, add select all to value
  if (
    !isSingleMode &&
    ensureIsArray(value).length === fullSelectOptions.length &&
    fullSelectOptions.length > 0
  ) {
    setSelectValue(
      labelInValue
        ? ([...ensureIsArray(value), selectAllOption] as AntdLabeledValue[])
        : ([
            ...ensureIsArray(value),
            SELECT_ALL_VALUE,
          ] as AntdLabeledValue[]),
    );
  }
}, [value, isSingleMode, labelInValue, fullSelectOptions.length]);
```